src/lib.rs

```diff
@@ -50,18 +50,37 @@ pub mod ahp;
 pub use ahp::AHPForR1CS;
 use ahp::EvaluationsProvider;
 use algebra_utils::get_best_evaluation_domain;
+use crate::ahp::verifier::VerifierState;
+use crate::ahp::prover::ProverMsg;
 
 #[cfg(test)]
 mod test;
 
+/// Configuration parameters for the Marlin proving system, modifying the
+/// internal behaviour of both prover and verifier.
+pub trait MarlinConfig: Clone {
+    /// If set, an optimization exploiting LC of polynomials will be enabled.
+    /// This optimization allows to include less opening values in the proof,
+    /// thus reducing its size, but comes with additional variable base scalar
+    /// multiplications to be performed by the verifier in order to verify the
+    /// sumcheck equations, that are expensive to circuitize.
+    /// If unset, all the opening values of all the polynomials are included in
+    /// the proof (thus increasing its size), but doesn't come with any additional
+    /// operation to be performed by the verifier, apart from the lincheck of course.
+    const LC_OPT: bool;
+    /// Enable or disable zero-knowledge
+    const ZK: bool = true;
+}
```

Comment:
Let's add a : `const SEGMENTATION: Option<usize>` in the config.
If not specified no segmentation is used, if specified segmentation shall be used with the specified segment size.
The segmentation is kind of transparent to the Marlin prover/verifier, but in our case we may allow Sidechains to use different segment sizes for their DLOG proving/verification keys, but all these keys shall be derived from a single DLOG proving/verification key declared in Mainchain.
Therefore, what I would do is,  in the Marlin `prove `and `verify` functions, call the `PC::trim() ` function on the DLOG proving/verification key passing as argument the segment size (of course let's check before that, that the segment size is not bigger than the proving/verification key)

Reply:
Let's expand the exisisting unit tests with the segmentation case too